pkg/operator/controller/machineconfig/mcp.go

```diff
@@ -235,15 +226,6 @@ func (r *MachineConfigReconciler) checkWorkerMCPStatus(ctx context.Context) (ctr
 
 	r.Lock()
 
-	if r.CtrlConfig.Status.IsDebuggingEnabled() && !r.CtrlConfig.Status.IsDebuggingFailed() {
```

Comment:
Condition handled [here](https://github.com/openshift/node-observability-operator/blob/84ffe4e80f367bfd2e03b6073392375b7f0feb1b/pkg/operator/controller/machineconfig/controller.go#L426), while calling the method, hence removed redundant check.